syntax = "proto3";

package khaleesi.core.clocktower;

import "core.proto";


message Job {
  string id          = 1;
  string name        = 2;
  string description = 3;

  string                               cron_expression       = 10;
  string                               action                = 11;
  khaleesi.core.JobActionConfiguration action_configuration  = 12;

  bool                                        is_cleanup_job        = 20;
  khaleesi.core.JobCleanupActionConfiguration cleanup_configuration = 21;
}

message JobRequest {
  khaleesi.core.RequestMetadata request_metadata = 1;
  Job                           job              = 2;
}


service BellRinger {
  rpc CreateJob  (JobRequest)              returns (khaleesi.core.IdMessage);
  rpc ExecuteJob (khaleesi.core.IdRequest) returns (khaleesi.core.EmptyResponse);
}
